docs/tutorials/fsdp_and_shard_tensor.rst

```diff
@@ -0,0 +1,230 @@
+ShardTensor and FSDP Tutorial
```

Comment:
Would maybe put "Domain Decomposition" in the title here to highlight this is not just some other way of doing FSDP